lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4113,6 +4134,30 @@ namespace {
             func->setImportAsStaticMember();
           }
         }
+
+        if (importedName.isSubscriptAccessor()) {
+          assert(func->getParameters()->size() == 1);
+          const auto parameterType = func->getParameters()->get(0)->getType();
+          if (!parameterType)
+            return nullptr;
+
+          auto cxxSubscripts = Impl.CXXSubscripts[parameterType];
```

Comment:
Does this get confused if we try to import multiple subscripts with the same parameter type?

Reply:
Oh, you're right. I've missed the fact that for nested structs that have `operator[]` with the same parameter type there would be a collision here. Fixed by changing the key type of `Impl.cxxSubscripts ` & added a test with nested structs.